### PR TITLE
Integrate IdP templates API

### PR DESCRIPTION
**Pull request by tharindu-bandara**
## Purpose
- Addresses https://github.com/wso2/identity-apps/issues/354.
- **Needs to merge: https://github.com/wso2/identity-apps/pull/636 first**.

## Goals
- Remove hard-coded templates.
- Retrieve templates from API and sort.
- If the image of the retrieved template matches a pre-defined icon, use that.
- Handle empty templates scenario.
## Approach
- Templates are retrieved from the API and get sorted.
Screenshot from 2020-03-30 21-00-55
- Handle scenario where zero templates exist.
Screenshot from 2020-03-30 20-56-04

### CODE REVIEW

File: apps/admin-portal/src/api/identity-provider.ts [hunk 1]
```diff
@@ -19,13 +19,18 @@
...
+    ApplicationTemplateListInterface,
     FederatedAuthenticatorListItemInterface,
     FederatedAuthenticatorMetaInterface,
     HttpMethods,
     IdentityProviderInterface,
     IdentityProviderListResponseInterface,
-    IdentityProviderResponseInterface
+    IdentityProviderResponseInterface, IdentityProviderTemplateListInterface, IdentityProviderTemplateListItemInterface
 } from "../models";
+import {AxiosError, AxiosResponse} from "axios";
```
Comment (brionmario) [inline apps/admin-portal/src/api/identity-provider.ts]: Add a space between the import curly braces.

File: apps/admin-portal/src/pages/identity-provider-edit.tsx [hunk 2]
```diff
@@ -27,7 +27,7 @@ import { IdentityProviderConstants } from "../constants";
 import { useDispatch } from "react-redux";
 import { addAlert } from "@wso2is/core/store";
 import { AlertLevels } from "@wso2is/core/models";
-import { EditIdentityProvider } from "../components/identityProviders/identity-provider-edit";
+import { EditIdentityProvider } from "../components/identity-providers/identity-provider-edit";
```
Comment (brionmario) [inline apps/admin-portal/src/pages/identity-provider-edit.tsx]: Can we export this component from `components` index. Then we'll be able to shorten the import as follows.

```suggestion
import { EditIdentityProvider } from "../components";
```

File: apps/admin-portal/src/api/identity-provider.ts [hunk 3]
```diff
@@ -19,13 +19,18 @@
...
+import {ApplicationManagementConstants} from "../constants";
+import { IdentityAppsApiException } from "@wso2is/core/exceptions";
+import {IdentityProviderManagementConstants} from "../constants/identity-provider-management-constants";
```
Comment (brionmario) [inline apps/admin-portal/src/api/identity-provider.ts]: Shall we export `IdentityProviderManagementConstants` file from the constants folder index.

File: apps/admin-portal/src/api/identity-provider.ts (hunk 1 above)
Comment (tharindu-bandara) [reply apps/admin-portal/src/api/identity-provider.ts]: Fixed with a71ec75bd7ff9034c30a22f8fd552a5d61bc892e.

File: apps/admin-portal/src/pages/identity-provider-edit.tsx (hunk 2 above)
Comment (tharindu-bandara) [reply apps/admin-portal/src/pages/identity-provider-edit.tsx]: Fixed with a71ec75bd7ff9034c30a22f8fd552a5d61bc892e.

File: apps/admin-portal/src/api/identity-provider.ts (hunk 3 above)
Comment (tharindu-bandara) [reply apps/admin-portal/src/api/identity-provider.ts]: Fixed with a71ec75bd7ff9034c30a22f8fd552a5d61bc892e.

**Pull request merged**